processor/processor.go

```diff
@@ -202,10 +202,14 @@ func (proc *HandleT) newUserTransformationStat(sourceID, workspaceID string, des
 
 	tags["transformation_id"] = destination.Transformations[0].ID
 	tags["transformation_version_id"] = destination.Transformations[0].VersionID
+	tags["error"] = "false"
 
 	numEvents := proc.statsFactory.NewTaggedStat("proc_transform_stage_in_count", stats.CountType, tags)
 	numOutputSuccessEvents := proc.statsFactory.NewTaggedStat("proc_transform_stage_out_count", stats.CountType, tags)
-	numOutputFailedEvents := proc.statsFactory.NewTaggedStat("proc_transform_stage_error_count", stats.CountType, tags)
+
+	errTags := tags
```

Comment:
by assigning a map to a new variable you are copying the map header but not the underlying bucket arrays.
https://go.dev/play/p/SrTn16f_-jy

You need to either create a new map and copy all elements of one map to another, or 

```
errTags := buildStatTags(sourceID, workspaceID, destination, USER_TRANSFORMATION)
errTags["transformation_id"] = destination.Transformations[0].ID
errTags["transformation_version_id"] = destination.Transformations[0].VersionID
errTags["error"] = "true"
```

_Note:_ same comment applies for all other scenarios where an `errTags` is prepared in this pull request

Reply:
wrote a new func to create a copy of map and used it.